kernel/src/main/java/org/kframework/compile/MarkExtraConcreteRules.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) K Team. All Rights Reserved.
+package org.kframework.compile;
+
+import org.kframework.attributes.Att;
+import org.kframework.definition.Definition;
+import org.kframework.definition.DefinitionTransformer;
+import org.kframework.definition.Rule;
+import org.kframework.utils.errorsystem.KEMException;
+
+import javax.annotation.Nullable;
+import java.util.HashSet;
+import java.util.List;
+
+public class MarkExtraConcreteRules {
+
+    /**
+     * Mark with [concrete] rules with labels enumerated in `--concrete-rules`.
+     */
+    public static Definition mark(Definition def, @Nullable List<String> extraConcreteRuleLabels) {
+        if (extraConcreteRuleLabels == null) {
+            return def;
+        }
+        HashSet<String> concreteLabelsSet = new HashSet<>(extraConcreteRuleLabels);
+        Definition result = DefinitionTransformer.fromSentenceTransformer((mod, s) -> {
+            if (s instanceof Rule) {
+                Rule r = (Rule) s;
+                String label = r.att().getOption(Att.LABEL()).getOrElse(() -> null);
+                if (label != null && extraConcreteRuleLabels.contains(label)) {
+                    concreteLabelsSet.remove(label);
```

Comment:
Is it possible to have multiple rules with the same label?
I would change this pass to add the used labels to a new collection and compare `!(concreteLabelsSet - usedLabels).isEmpty()`

Reply:
Ah, I just noticed `concreteLabelsSet` is a copy.
Then can you rename it to something like `unused`?
I think it makes more sense that way.

Reply:
Rule labels are supposed to be unique. I believe the backends assume this too. It should be an error to have multiple rules with same label.

Reply:
Ah good to know, I'll update the code accordingly. (I also double-checked, and an error is indeed reported if there are non-unique labels).